tools/offline_ota_image_builder/__main__.py

```diff
@@ -0,0 +1,165 @@
+        _parsed = str(raw_pair).split(":")
+        if len(_parsed) >= 2:
+            _ecu_id, _image_fpath, _image_version, *_ = *_parsed, ""
+            image_metas.append(
+                ImageMetadata(
+                    ecu_id=_ecu_id,
+                    image_version=_image_version,
+                )
+            )
+            image_files[_ecu_id] = _image_fpath
+        else:
+            logger.warning(f"ignore illegal image pair: {raw_pair}")
+
+    if not image_metas:
+        print("ERR: at least one valid image should be given, abort")
+        sys.exit(-1)
+
+    # ------ parse export options ------ #
+    output_fpath, write_to_dev = args.output, args.write_to
+
+    if write_to_dev and not Path(write_to_dev).is_block_device():
+        print(f"{write_to_dev} is not a block device, abort")
+        sys.exit(-1)
+
+    if write_to_dev and not args.confirm_write_to:
+        _confirm_write_to = input(
+            f"WARNING: generated image will be written to {write_to_dev}, \n"
+            f"\t all data in {write_to_dev} will be lost, continue(type [Y] or [N]): "
+        )
+        if _confirm_write_to != "Y":
+            print(f"decline writing to {write_to_dev}, abort")
+            sys.exit(-1)
+    elif write_to_dev and args.confirm_write_to:
+        logger.warning(
+            f"generated image will be written to {write_to_dev},"
+            f"all data in {write_to_dev} will be lost"
+        )
+
+    # ------ build image ------ #
+    with tempfile.TemporaryDirectory(prefix="offline_OTA_image_builder") as workdir:
+        build(
+            image_metas,
+            image_files,
+            workdir=workdir,
+            output=output_fpath,
+            write_to_dev=write_to_dev,
+        )
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO, format=cfg.LOGGING_FORMAT, force=True)
+    parser = argparse.ArgumentParser(
+        prog="offline_ota_image_builder",
+        description=(
+            "Helper script that builds offline OTA image "
+            "with given OTA image(s) as external cache source or for offline OTA use."
+        ),
+    )
+    parser.add_argument(
+        "--image",
+        help=(
+            "OTA image for <ECU_ID> as tar archive(compressed or uncompressed), "
+            "this option can be used multiple times to include multiple images."
+        ),
+        required=True,
+        metavar="<ECU_NAME>:<IMAGE_PATH>[:<IMAGE_VERSION>]",
+        action="append",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="save the generated image rootfs into tar archive to <OUTPUT_PATH>.",
+        metavar="<OUTPUT_PATH>",
+    )
+    parser.add_argument(
+        "-w",
+        "--write-to",
+        help=(
+            "write the image to <DEVICE> and prepare the device as "
+            "external cache source device."
+        ),
+        metavar="<DEVICE>",
+    )
+    parser.add_argument(
+        "--confirm-write-to",
```

Comment:
confirm-write-to seems to be an option that requires confirmation.
The default is confirm, and I think it is natural if --force is specified not to confirm.